src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -126,10 +129,17 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
         this.etLogEventType = etLogEventType;
         this.eventTypeOptionsValidator = eventTypeOptionsValidator;
         this.adminService = adminService;
+        this.authorizationService = authorizationService;
     }
 
     public List<EventType> list() {
-        return eventTypeRepository.list();
+        return authorizationService.filter(eventTypeRepository.list()
+                .stream()
+                .map(EventType::asResource)
+                .collect(Collectors.toList()))
```

Comment:
The indentation here is kind of trick to read. There is an additional `)` which mean the next call from the line bellow is on the result of the filter (the unwrap part). Maybe we can somehow change the indentation to make it more clear.